src/library/features/crates/cratetablemodel.cpp

```diff
@@ -11,19 +10,20 @@
 CrateTableModel::CrateTableModel(QObject* pParent,
                                  TrackCollection* pTrackCollection)
         : BaseSqlTableModel(pParent, pTrackCollection,
-                            "mixxx.db.model.crate") {
+                            "mixxx.db.model.crate"),
+          m_pCrates(pTrackCollection->crates()) {
 }
 
 CrateTableModel::~CrateTableModel() {
 }
 
-void CrateTableModel::selectCrate(CrateId crateId) {
+void CrateTableModel::selectCrate(Crate crate) {
```

Comment:
Why do you now pass the complete crate object (by value) if you still only need the id as before?